deploy/workflow/workflow.md

```diff
@@ -0,0 +1,20 @@
+# Proposed Workflow Changes
+
+As per recent discussion, here is an outline of the proposed changes...
+
+## Step 1: Developer starts work on feature/bug
+
+![work-starts](https://github.com/DFEAGILEDEVOPS/MTC/blob/cd3898d3cba00fd73f791341152b8a1abd8479c8/deploy/workflow/work-starts.png)
+
+## Step 2: Work in progress for dev & QA
+
+![work-in-progress](https://github.com/DFEAGILEDEVOPS/MTC/blob/cd3898d3cba00fd73f791341152b8a1abd8479c8/deploy/workflow/work-in-progress-review.png)
+
+## Step 3: Dev & QA work complete
+
+![dev-qa-complete](https://github.com/DFEAGILEDEVOPS/MTC/blob/a45c8dbf90136695dec669bfe30f32280af93553/deploy/workflow/dev-qa-complete.png)
```

Comment:
My main concern with this is that unless we run the e2e tests before merging in to dev, dev will simply become a permanently broken branch, making it hard to work from.

We will also have to manually investigate test failures on the dev->master merge and guess which PR merge to dev caused the issue to assign it to someone.

The fact that this is a manual investigation is enough for me to not be in favour of it.

Have I missed something here?

Reply:
`dev will simply become a permanently broken branch, making it hard to work from`  we _could_ create branches from master.  we should explore this idea.

`We will also have to manually investigate test failures on the dev->master merge` we are doing this anyway.

`... and guess which PR merge to dev caused the issue to assign it to someone.` it should be obvious.  QA and Dev should both have working tests before the merge to master happens.